### ISSUE TITLE
Testing autopublish.meteor.com

### ISSUE THREAD

**Issue by splendido** [MeteorPackaging/autopublish.meteor.com#1]

**Comment by dandv** [comment]
Nice UI!

**Comment by jlukic** [comment]
:+1:

**Comment by splendido** [comment]
Hi @jlukic,
seeing your thumb up here is a real pleasure!

...please feel free to advice about anything you might notice you would have done differently ;-)

**Comment by splendido** [comment]
Thank you @rgoomar and @zimme for testing this out!
...and it seems everything went good at the first time :)

Anything you'd like to change/add/report?

**Comment by zimme** [comment]
No problem @splendido. I've set it up for all my packages. I have a couple of thoughts.

How do we handle .versions file now that we don't publish manually i.e. the .versions file won't get updated. How I handle it right now is to just not check in a .versions file into git.

Should autopublish maybe publish old versions when you add a project and there isn't anything published yet? Maybe it already does this?

**Comment by zimme** [comment]
Also please update the description of `meteorpublish` user/org on atmosphere =)

**Comment by splendido** [comment]
@zimme: it [the description](https://atmospherejs.com/meteorpublish) any better now?
feel free to suggest more changes!

**Comment by splendido** [comment]
...for the `.versions` I agree not to push them to the repositories: they will be recreated from scratch by `meteor publish`

as for publishing old versions, it might be tricky since checking out old tags wouldn't preserve the `pacakge.js` file.
Moreover you should publish older version first and it is not guaranteed that the current `package.js` file would work for old versions too.

That should be a manual process I'd say...

Thoughts?

**Comment by zimme** [comment]
Looks good :+1:

'Bot for automatic package publishing' might be a little better?

**Comment by zimme** [comment]
That makes sense, I think it needs to be manual

**Comment by splendido** [comment]
description for `meteorpublish` updated ;-)

**Comment by sevki** [comment]
I can't log in :disappointed:

**Comment by splendido** [comment]
what's going on @sevki?
...any errors in the console???

**Comment by sevki** [comment]
tried it a bunch of times but only a oauth redirect with a blank page would return. it seems to be fixed now tho

**Comment by sevki** [comment]
![autopublish_meteor_com__oauth_github_close__code_7e3bc6c5eee867dc301d_state_eyjsb2dpbln0ewxlijoicg9wdxailcjjcmvkzw50awfsvg9rzw4ioijrd3vfwgzkvnric1dty1g3swxputrpltzamthbr3byzjfgq0fkafj4ddbjiiwiaxndb3jkb3zhijpmywxzzx0_3d_and_meteor_autopublis](https://cloud.githubusercontent.com/assets/429977/7001872/2bd5bd02-dc4a-11e4-9c42-13621f085890.png)

**Comment by splendido** [comment]
@sevki, it's working for me! :-(

is the screenshot taken after you submitted the form?
...from the first part of the url it seems it should be closed at that time...

It would be great if you could so some separate tests with the `Meteor.loginWithGitHub`...
There's nothing I could actually do with this, I'm just using the Meteor API for the accounts, not even my `useraccounts` packages...

**Comment by deanrad** [comment]
Looks intriguing, I'll try it out with octokat.js

**Comment by splendido** [comment]
@chicagogrooves: Let us know your feedback ;-)

**Comment by trusktr** [comment]
Does it work with binary builds yet?

**Comment by splendido** [comment]
@trusktr at the moment there's nothing to autodetect binary packages, but we could (manually?) mark some subscritpion to be published for all architectures (it's just a matter to cycle through available build machines and running `meteor publish for arch` instead of `meteor publish`, right?).

Let me know in case you're interested to test it with some package, we'll sort something out!

**Comment by trusktr** [comment]
@splendido Yeah! This feature on autopublish.meteor.com would an awesome time saver! I'm developing `rocket:module` right now, and it requires binary builds, and the process is time consuming to do manually. I would totally test it with rocket:module.

**Comment by splendido** [comment]
any new release coming soon?

**Comment by trusktr** [comment]
Well, I've been testing things out locally, and I posted some 0.x.x releases on Atmosphere in order to test the publish build process (for example, publishing a package relying on rocket:module to see how the process works, and I've learnt a lot that I otherwise wouldn't have). I didn't know if there was a way to test publishing without actually publishing to Atmosphere. I plan for `rocket:module@1.0.0` to be the actual working release. I haven't needed to release anything recently since I haven't needed to test publish builds recently, and I might not have to anymore. I'm down to publish (possibly non-working) 0.x.x versions just to test autopublish.meteor.com.

**Comment by splendido** [comment]
Ok, let me know when you create your subscription and I'll try to modify the publish procedure to see what happens...
There might be something to change while you try it, so please be aware there might be something odd appening (mostly on the autopublish.meteor.com reporting of publishing operations than for actual publish commands though...)

If you're available for testing, please subscribe `rocket:module`! :-)

**Comment by trusktr** [comment]
@splendido I think I've done it. (: I'm not sure if anything has happened.

**Comment by splendido** [comment]
I had to _approve_ your hook and subscription first...
So I guess the trigger received by autopublish was ignored ;-)

Please give me a bit of time to review the publish process to cycle through available architectures...

**Comment by trusktr** [comment]
@splendido Aawwesomme. Do build errors get reported somewhere?

**Comment by splendido** [comment]
yes, you can show a log like [this](https://autopublish.meteor.com/log/sdMA5zGQ4M4qcddbZ), but in this case I'm thinking we should get four different logs, one for each architecture...
...and, also, the result might not be just failure/success but there might be some success and some failures...
Lets see...

**Comment by splendido** [comment]
@trusktr, just a stupid question:
should a regular `meteor publish` be run before the others `meteor publish-for-arch`?

**Comment by trusktr** [comment]
@splendido When you try to `meteor publish` a package with binary builds, you'll get a message like this one with instructions on what to do next: https://forums.meteor.com/t/automatically-publishing-a-meteor-package-with-binary-code/3637

**Comment by trusktr** [comment]
I added the meteorpublish user to the rocket organization. Do you have access to that user? If so you can clone https://github.com/trusktr/rocket-module and try `meteor publish`. ~~I can add you as maintainer to rocket:module too. Let me know.~~ I added you as maintainer.

**Comment by splendido** [comment]
ok, I've seen it. See also #24
But so, the first `meteor publish`is not necessary? Could we run directly `meteor publish-for-arch` without the first `meteor publish`?
I might be ready in a minute if you'd like to test it out...
...I'm going to trigger four different publish, each one for a different architecture, so that we can get a log for each publish operation.
The only problem would be retriggering just some out of the four in case something go wrong...
At the moment only _admin_ users see a button to re-trigger a publish operation, but since the code for a particular release cannot be changed (or could it??) I think there is no point in letting a user re-trigger a failing publish...

Having `meteorpublish` in the `rocket` organization is enough, thanks!

**Comment by trusktr** [comment]
Well, the thing is that `meteor publish-for-arch` has to be run in the OS that the binary is compiled for, so, for example, you have to run `meteor admin get-machine os.windows.x86_32` which puts SSHes you into a build machine hosted by Meteor, then in that machine you have to run `meteor publish-for-arch`. You have to run `meteor publish-for-arch` in each machine.

**Comment by splendido** [comment]
Could you try to make a new release?
I'd say everything's ready for the first automated publish for a binary package!
...this would make a bit of story for the Meteor community! Wouldn't it? ;-)

**Comment by trusktr** [comment]
Yeah, once `meteor publish` has been run, that source code has been sent to Meteor/Atmosphere, and then any time you run `meteor publish-for-arch` in any machine, it builds based on that submitted source code which cannot be modified (I tried that :laughing:). The only solution is to `meteor publish` another version as far as I know.

It'd be cool if (since the package requires a binary build) It'd let you change the source until at least one successful build, to fix errors. <-- @glasser

**Comment by trusktr** [comment]
Yeah, this would be a huge thing for publishing!!

**Comment by splendido** [comment]
so, wait!
`meteor publish` first or not?!?!
I'm sorry I got lost! :(

**Comment by trusktr** [comment]
Yeah, `meteor publish` first, then get a machine for each arch, then in each machine run `meteor publish-for-arch`.

**Comment by splendido** [comment]
So lets wait another minute please...

**Comment by trusktr** [comment]
K, I pushed a new version of rocket:module to github.

**Comment by trusktr** [comment]
Should I see the process initiated in the "Queueing Publish" box of the home page?

**Comment by splendido** [comment]
In principle yes...
Let me check!

**Comment by splendido** [comment]
Everything seem's fine on the autopublish.meteor.com side, but it got only the first trigger to test the hook you created...
No further triggers after then!

Could you please check at the bottom of [this page](https://github.com/trusktr/rocket-module/settings/hooks/5060848) how many 'Recent Deliveries` you see?
(I have no access to the page, obviously...)

**Comment by splendido** [comment]
...also, I'd like to make a recording of the screen while you're packages wil be being published...
Could you please let me know when you're making a new release again?

(edit: or you could do the screen recording yourself ;-) )

**Comment by splendido** [comment]
Btw, the settings for the hook should lool like [this picture](https://github.com/MeteorPackaging/autopublish.meteor.com/wiki/How-to-manually-set-up-web-hooks#releases)

**Comment by trusktr** [comment]
Aha, `tags` wasn't selected. I deleted a tag and re-pushed it, now I see a new payload sent.

**Comment by trusktr** [comment]
Cool, it showed up in the site. It errored. Getting closer. :D

**Comment by trusktr** [comment]
Oh, let me try the newer version.

**Comment by splendido** [comment]
yeah, the log says:
`error: No compatible binary build found for this package. Contact the package
author and ask them to publish it for your platform.`

But it should have scheduled five publishes... :(

**Comment by splendido** [comment]
Oh, wait.
You said you're using tags and not releases?

**Comment by trusktr** [comment]
Yeah

**Comment by splendido** [comment]
mmm, but I can see two new releases [here](https://github.com/trusktr/rocket-module/releases)

**Comment by trusktr** [comment]
Hmmm, I think GitHub makes those automatically based on the tags I've pushed.

**Comment by trusktr** [comment]
I think it recognizes `vX.X.X` tags.

**Comment by splendido** [comment]
yep, it was a tag hook!
I have to add another bit of code to schedule accordingly also for tags...

...but: **why is `meteor publish` failing**?

**Comment by trusktr** [comment]
Oh, hmmm, let me try it out here and see.

**Comment by trusktr** [comment]
Oops, I was testing some code locally, but it isn't meant to run during publish. doh :laughing: Let me fix and try that again.

**Comment by splendido** [comment]
It should be ready also for tags now!

**Comment by splendido** [comment]
at the next new tag, you should see five different publishing sheduled:
- `meteor publish` first
- `meteor publish-for-arch` four times later on

**Comment by trusktr** [comment]
Okay, giving it a go!

**Comment by splendido** [comment]
I'm recording! :-)

**Comment by splendido** [comment]
:(

**Comment by trusktr** [comment]
Hehe. Perhaps the binary ones can be labeled like "rocket:module (os.linux.x86_64)", etc.

**Comment by splendido** [comment]
yep, that would be an improovement ;-)

**Comment by splendido** [comment]
I'm now curious to see what happened...
Build machines seem to be correctly taken and connected to, btw

**Comment by trusktr** [comment]
> error: No compatible binary build found for this package. Contact the package
> author and ask them to publish it for your platform.

I wonder what that means. Hmmmmm.

**Comment by trusktr** [comment]
:open_mouth: One succeeded! :smile:

**Comment by splendido** [comment]
Did you ever get something like it locally?
...could it be something related to using a different user?

What I do now is:
- get a build machine
- login `meteorpublish` with `meteor login`
- git clone
- `meteor publish` or `meteor publish-for-arch`

**Comment by trusktr** [comment]
Which architecture what that for? I think I might know what's happening. Maybe while building for certain arch, a dependency for rocket:module can't be found in that arch. I might have to use a different tool!

**Comment by trusktr** [comment]
Oh!!!! Its rocket:webpack. :laughing:

**Comment by trusktr** [comment]
Now I can add rocket:webpack to autopublish.meteor.com. :smile:

**Comment by splendido** [comment]
funny enough ;-)

**Comment by splendido** [comment]
I'll approve it and mark it as binary asap!

**Comment by trusktr** [comment]
Would've been great if the error mentioned "rocket:webpack". Are you able to see if it does?

**Comment by splendido** [comment]
Everything's ready for `rocket:webpacket`!

...the log we can see on the browser is all I'm able to retrieve at the moment

**Comment by splendido** [comment]
mmm, actually this is the last part of the log string:

```
\\\r=> Errors while initializing project:\r   Selecting package versions                \\\rWhile checking for rocket:webpack@1.9.10_1:\nerror: No compatible binary build found for this package. Contact the package\nauthor and ask them to publish it for your platform.\r   Selecting package versions                \\
```

I might have some problem with `\r`s

**Comment by splendido** [comment]
I spoke too early!
If you got to the right end of the log textarea, you can see it!

What would you suggert to show the log in a more intelligible way??? :(

**Comment by trusktr** [comment]
Hmmm, I didn't realize the text didn't wrap. If it wrapped I would've seen the error. I suppose preventing the text from scrolling to the side would be great (like happens in terminals)!

**Comment by splendido** [comment]
First one done!

...I'm adding the detail about the architecture for the items being published.
If you could wait a bit before making a new tag for `rocket:module` we should get a really good video ;-)

**Comment by trusktr** [comment]
This is sooo awesome!!! wooooo!!! :+1:

**Comment by splendido** [comment]
...but one failed ;-)

**Comment by trusktr** [comment]
Ooops, "version already exists". :laughing: haha

**Comment by trusktr** [comment]
Will try again.

**Comment by splendido** [comment]
No, that's my fault actually!
It seems it's running `meteor publish` everytime and not `meteor publish-for-arch`!!!

**Comment by splendido** [comment]
I've changed a bit the selection of the publish command and I'm going to manually retrigger your tags.
I'll see if the first one succeeds...

**Comment by splendido** [comment]
Ok, we're getting there...
See [this log](https://autopublish.meteor.com/log/p74B4k7teHePMj6Sd)

It seems you don't need a copy of the code when you run `meteor publish-for-arch` and you also have to specify `packagename@version`

...I'm sorry if you already told me and I didn't pick it up.

I'll have to change strategy a bit now...

**Comment by trusktr** [comment]
> It seems you don't need a copy of the code when you run meteor publish-for-arch and you also have to specify packagename@version

Yep! It will use the code submitted during the original `meteor publish` command, which only needs to run once. (:

**Comment by trusktr** [comment]
Might it be helpful if the successful ones (green) can also be clicked to show the log (just no popup)?

**Comment by trusktr** [comment]
Let me know when to try the next version. :)

**Comment by splendido** [comment]
Deployed!
Could you be so kind to release a  new version of `rocket:webpack`?

**Comment by splendido** [comment]
Finger crossed! :-)

**Comment by trusktr** [comment]
hehe :)

**Comment by trusktr** [comment]
Is it doing one task now instead of 5?

**Comment by splendido** [comment]
If this one succeeds, the others will be scheduled later on...

**Comment by trusktr** [comment]
sweeeet!

**Comment by splendido** [comment]
I have to admin that coding without having the possibility to test it locally might be _difficult_ sometimes...
But if I got it right at first shot, I'll go to bed really happy :-)

**Comment by trusktr** [comment]
hehe yeah!! Exactly what I thought about publishing to Atmosphere. The only way to test some scenarios is to actually publish things, which means we might publish junk versions.

**Comment by splendido** [comment]
actually, atmosphere is just a show off of the packages DB hosted by MDG.
But you're right, that's the truth.

I somewhere asked for a `meteor publish --dry-run` somewhere in the past, and a few people agreed that would make sense to have something to test your publish before actually publish.
...we're still at the same point tough!

**Comment by splendido** [comment]
I'm sorry man!
this time I run `~/.meteor/meteor publish-for-arch rocket:webpack@1.9. 
10_3`
(there's a `\n` in between `1.9.` and `10_3`...)

...this f....n' log full of `\r`s and `\n`s!!!! Grrrrrrrr.....

Let me improove a bit the parsing of the successfully published version :(

**Comment by trusktr** [comment]
Ooof, so many `\   Building package rocket:webpack` animations. hehe.

**Comment by splendido** [comment]
Yeah, it's a pain...

I'd like to find out a good regexp to get rid of all that noise....
I once tried with [this function](https://github.com/MeteorPackaging/autopublish.meteor.com/blob/cad2b7c8f0909cb73473b40ab0e4467ecdf974f4/server/migrations/strip_logs.js#L7-14) but the result seemed to be missing something... :(

**Comment by splendido** [comment]
I'm lost!
The version retrieved from the successful publish seems to  be correct: `1.9.10_3`
and the command run later on seems to be correct too: `~/.meteor/meteor publish-for-arch rocket:webpack@1.9.10_3`

But if you look at the log reported on the browser you see: `~/.meteor/meteor publish-for-arch rocket:webpack@1.9. 10_3` and, whatmore, if you copy/paste it you see a new line in the middle :-(

**Comment by trusktr** [comment]
I saw that. Strange!

**Comment by trusktr** [comment]
Is it `~/.meteor/meteor publish-for-arch rocket:webpack@1.9.`?

**Comment by splendido** [comment]
mmm, perhaps it's just a problem with the log retrieval:

~/.meteor/meteor whoami\nmeteorpublish\nrm -rf autopublish\n~/.meteor/meteor publish-for-arch rocket:webpack@1.9. \r10_3\r   Updating package catalog                  |\r   Updating package catalog

Do you have any idea about why it could be failing?

**Comment by splendido** [comment]
The only meaningful part seems to be `error: Cannot override existing build`

**Comment by trusktr** [comment]
https://github.com/meteor/meteor/issues/4477

**Comment by trusktr** [comment]
Ironically, I posted that issue. x)

**Comment by splendido** [comment]
...I see :)

So what?
Was it published or not?!
...should we try publishing `rocket:module` then?

**Comment by splendido** [comment]
I've read it better. So it seems it all depends on which architecture you choose for the first publish...
Am I correct?

In fact there's no trace within the log of the first successful publish operation of [this warning](https://github.com/meteor/meteor/blob/devel/tools/commands-packages.js#L463-465)!

...do you think we should add a field to specify where to execute the first `meteor publish'?

**Comment by trusktr** [comment]
https://github.com/meteor/meteor/issues/4477#issuecomment-111776380

**Comment by trusktr** [comment]
It doesn't matter where we run `meteor publish`, since that just uploads the source (JavaScript runs everywhere). What @glasser is saying is that the binary build was canceled (although it threw an error) and it should be published as platform-agnostic, but this doesn't seem to be the case because rocket:module is complaining it can't find certain platform builds.

Let me give rocket:module another shot.

**Comment by trusktr** [comment]
What do you think about changing "Version" to "v" so long versions don't push the time-ago box down?

![screen shot 2015-06-13 at 8 21 41 pm](https://cloud.githubusercontent.com/assets/297678/8146934/f6095f58-1209-11e5-943a-f94cd7d90578.png)

vs

![screen shot 2015-06-13 at 8 24 01 pm](https://cloud.githubusercontent.com/assets/297678/8146940/271207c6-120a-11e5-841a-b63d43707080.png)

**Comment by splendido** [comment]
....mmm I understand it differently:

> So when you build this package on a Mac, you get fsevents and Meteor detects that there is binary code here and publishes it as a Mac-specific package.
> 
> When you build it on other platforms, fsevents fails to install, but that's OK since it's an optional dependency. So it goes and publishes it as a platform-agnostic package.

This means to me that if you run `meteor publish` for the first time on MAC, the presence of `fsevents` make meteor think it's a binary packages and publishes it as _platform specific_. Which I guess it means you should have no problems when you publish it on a different machine later on...

If, on the other side, you publish on linux first, no binary dependency is detected and hence the package is published as _platform agnostic_ for good!

Does this make any sense?

**Comment by splendido** [comment]
...I'll change `Version` to `v`, thanks!

**Comment by trusktr** [comment]
The errors don't show which platform the package was being built for. Maybe the details list can include a "Platform" field?

**Comment by trusktr** [comment]
It's obvious which one is Windows. :laughing:

**Comment by trusktr** [comment]
> Does this make any sense?

If that were true, then running `meteor publish` wouldn't be platform-agnostic itself. I see what you mean though. If I run `meteor publish` for the first time in Linux, perhaps the binary build necessity won't be detected, which wouldn't be good! `meteor publish` should detect binary build necessity no matter which platform `meteor publish` runs on, otherwise if I were in Windows or Linux I would never have known I need to binary build for Mac OS X. @glasser Any insights here?

**Comment by trusktr** [comment]
What OS are you in @splendido? I don't have Linux or Window at hand at the moment to test with.

**Comment by splendido** [comment]
I think that is exactly what glasser meant with

> I think there's no actual bug here, even though it's confusing. I've added a commit that extends the error message to include the architecture it was trying to upload. (Also, in this odd case, it's unclear to me if we're guaranteed to choose the Mac build rather than the agnostic build when you're on a Mac and downloading packages.)

But I agree with you this is potentially a uge problem for developers publishing packages like your one!

**Comment by splendido** [comment]
this is the result for the lates successful publish:

Published rocket:module@0.1.5_1.

Nothing about being binary!
...the first arch I use it linux x64, that might be the reason.

I've just put an option to select the arch for the first publish.
...could we have another go with `rocket:webpack`? :-))

**Comment by splendido** [comment]
well, wait one sec for the deploy to finish...

**Comment by splendido** [comment]
do you think I should also increase the timeout (because of Windows...)?

**Comment by splendido** [comment]
Done!
Lets try again with a new version for `rocket:webpack` :-)

**Comment by splendido** [comment]
If I had it correct and the first publish will run on `os.osx.x86_64`, I'm almost sure we're done! :-)

**Comment by splendido** [comment]
Yup! :+1:

**Comment by trusktr** [comment]
> I've just put an option to select the arch for the first publish.

Could you elaborate on that?

**Comment by splendido** [comment]
"You're not done publishing yet! This package contains binary code and must be..."

I was right!!! :-)

**Comment by splendido** [comment]
...when I pick up the next publish operation form the DB I now check for the presence of a `firstArch` field.
In case it's present, I use the specified architecture to run `meteor publish` for the first time.
Differently I run it on linux x64

**Comment by splendido** [comment]
win win win!!!

**Comment by trusktr** [comment]
Check out that single error. I got that too when publishing manually. I think it might be because the binary dependency for the 32bit linux satisfies the 64 bit linux, with the same misleading `error: Cannot override existing build`.

**Comment by splendido** [comment]
Now it seems it makes no difference from `os.linux.x86.32` and `os.linux.x86.64`: the first one succeeded while for the second one it complains as before....

Yeah, I was writing the same ;-)

**Comment by trusktr** [comment]
I think if perhaps we ran for 64bit first, then we'd get the same error on 32bit. It'd be nice if it weren't an error, just a message.

**Comment by splendido** [comment]
What about windows? Have you ever tried to get a windows machine?
I'll have to dig more and see where the timeout is coming from...

**Comment by trusktr** [comment]
Let me try it manually right now and see. I think I was getting the same timeout error.

**Comment by splendido** [comment]
I'm in...
But I'm realizing now I usually run `~/.meteor/meteor`
...that might not work on WIN!

**Comment by trusktr** [comment]
Hehe, nope! It would be nice if instead of CMD they gave us GitBash (msysgit)!

**Comment by splendido** [comment]
oopps!
`rocket:module` was first run on linkux instead of mac...
Let me check the db

**Comment by trusktr** [comment]
Are you running `meteor publish` for the first time inside of a machine? Linux 64bit by default?

**Comment by trusktr** [comment]
Where do we find the package folder in the machine?

**Comment by trusktr** [comment]
Oh, you're git cloning them!

**Comment by splendido** [comment]
are you sure you want to know the sequence?! ;-)
1. from the meteor server running autopublish.meteor.com I login into a droplet of mine on DO to run `meteor admin get-machine xxx` and get the machine specs as a JSON object
2. then I get back to the server and log into the build machine..
3. repeat point 2. a number of time in case of binary pacakges ;-)

**Comment by trusktr** [comment]
SO what's the "first run"? Is that inside a machine, and you're running `meteor publish`?

**Comment by splendido** [comment]
yes, I'm already into a meteor build machine.

**Comment by trusktr** [comment]
Are you using the first run to detect if the package needs a binary build?

**Comment by trusktr** [comment]
Wanna get on gitter? We can chat faster there. I have some ideas.

**Comment by splendido** [comment]
are you on hangout or some slack team?

**Comment by trusktr** [comment]
I'm on hangouts. My email is trusktr@gmail.com. I'm also in famous-community.slack.com.

**Comment by splendido** [comment]
not yet!
...I've marked them by hands for now, but it should easily doable by intercepting the warning on the console :-)

**Comment by trusktr** [comment]
@splendido How might I add a new org to My Accounts? I've granted access to the app for the org on github.com already.

**Comment by splendido** [comment]
I think you have to check you're a "public" member of the org...
Lets check [this page](https://github.com/orgs/Buzztrax/people), e.g., and make sure you're not a private member.

**Comment by funkytaco** [comment]
I'd like to try this. Thanks. My org is CovertLamp

**Comment by trusktr** [comment]
@splendido Awesome, it worked! I'll be publishing rocket:module now. :D

**Comment by splendido** [comment]
At the moment we're experiencing some problems with our bot user not being able to log in.
We're already in touch with MDG to solve the problem.

I'll let you know when we'll able to restart the service.

Luca

**Comment by dandv** [comment]
Has the bot user problem been solved, now that `publishbot` has replaced the previous user?

Also, does the screencast need an update including `autopublish.json`?

**Comment by zimme** [comment]
I think that the bot has only been renamed actually, I haven't heard anything that suggests that the bot user is somewhat different now than before. I saw some issue thread about the bot user being allocated more build time on the meteor servers, but I think that was an older thread.

**Comment by jdlubrano** [comment]
Hey @splendido,

@dandv recently added me to the _nvd3_ organization.  I would like to give this project a try for the `nvd3:nvd3`.  In the next couple of days I plan to follow your templates for wrapping third-party libraries and apply the necessary changes to https://github.com/MeteorPackaging/nvd3.

Thanks!

**Comment by splendido** [comment]
these are great news @jdlubrano!
what if we create a new repository called `nvd3-wrapper` so that we can keep all names for wrapper repos ending with `-wrapper`?
...I'll create the new one right now ;-)

**Comment by splendido** [comment]
@funkytaco you should be good to go now.
Feel free to test publishing your packages with autopublish.meteor.com

**Comment by benjick** [comment]
Meep meep: https://autopublish.meteor.com/log/XNZA8WNvzoAMyAwzN

**Comment by deanrad** [comment]
Hey there - I'm now officially trying to integrate this library: https://github.com/deanius/dependito/tags.

Of course, I didn't fully read the instructions until afterwards, and while I've added `packagebot` as a contributor to the meteor admin, I didn't create a fork in the MeteorPackaging org  - is that still necessary ? Just checking to see if all the steps in the documentation are still necessary. 

If so, I'll start from the top and do things the right way :)

**Comment by splendido** [comment]
@benjick I'm sorry the build machine had very little free memory :(
...but now the error is different: see the [new log](https://autopublish.meteor.com/log/XNZA8WNvzoAMyAwzN).
I'd say `packagebot` was not added to the package's maintainers. is this correct?

**Comment by splendido** [comment]
@deanius if you're trying to create a wrapper package for _dependito_ you should have a look at [this page](https://github.com/MeteorPackaging/autopublish.meteor.com/wiki/Wrapping-3rd-Party-Packages).
I know it might be hard to follow since we still have to harmonize all the docs and there's still old stuff around, but please try to forgive us ;-)

**Comment by deanrad** [comment]
Thanks @splendido I'll take a look, no sweat its worth the work :)

**Comment by benjick** [comment]
@splendido oops, my bad! Guess I never used autopublish on this repo before. Thanks!

**Comment by sharlayoub** [comment]
Nice work

**Comment by trusktr** [comment]
Greetings @splendido. :]

I don't know what changed, but ever since Meteor 1.2 (don't know if that's coincidence or not), I haven't needed to do arch-specific builds of `rocket:module`, so I haven't really had a need to look into autopublish because `rocket:module` is really the only thing I'm repeatedly publishing to Atmosphere and `meteor publish` is quick enough now without arch-specific builds.

With [ES2016 modules coming out soon](https://forums.meteor.com/t/ideas-from-mdg-meteor-1-3-the-meteor-guide-and-es2015-modules/13591), I'll probably end up publishing most things to NPM.

**Comment by mathiasrw** [comment]
The Library [AlaSQL](http://github.com/agershun/alasql) would love to be part of your alpha

**Comment by splendido** [comment]
Unfortunately autopublish.meteor.com is no more reacheable.
As many of you might already know, *.meteor.com is no more available for deploying demo apps.

Since the last release of Meteor (v1.3) there's official support for ES2016 modules, so I guess this means there's no more need for things like `autopublish.meteor.com`.

**Comment by dandv** [comment]
@splendido: thank you so much for all your work. Even though autopublish is no longer needed, I believe it played a crucial role in putting pressure on the core Meteor team to make it possible to use npm modules directly.

Just like other shims, by becoming unnecessary, autopublish ultimately proved to be successful.

**Comment by splendido** [comment]
Thank you @dandv for the kind words, really appreciated!

...but don't forget the initiative about official wrapper packages was sopported by you essentially! ;-)
I think it's been a great experience, both for the project itself and for the community collaboration and cooperation!
